### PR TITLE
chore: add v1 proxy tests for salesforce 

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## What are the changes introduced in this PR?

Add v1 proxy tests for Salesforce

## What is the related Linear task?

Resolves INT-1492

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

File: test/integrations/destinations/salesforce/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+            ],
+          },
+        },
+      },
+    },
+  },
+  {
+    id: 'salesforce_v1_scenario_2',
+    name: 'salesforce',
+    description: '[Proxy v1 API] :: Test with invalid session id',
```
Comment (utsabc) [inline test/integrations/destinations/salesforce/dataDelivery/business.ts:125]:
```suggestion
    description: '[Proxy v1 API] :: Test with session expired scenario',
```

File: test/integrations/destinations/salesforce/dataDelivery/other.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { generateProxyV1Payload } from '../../../testUtils';
+import { externalIDSearchedData, reqMetadataArray, proxyMetdata } from './business';
+
+export const otherSalesforceScenariosV1 = [
+    {
+        id: 'salesforce_v1_scenario_7',
```
Comment (utsabc) [inline test/integrations/destinations/salesforce/dataDelivery/other.ts:6]:
```suggestion
        id: 'salesforce_v1__other_scenario_1`,
```
and so on

File: test/integrations/destinations/salesforce/dataDelivery/other.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        name: 'salesforce',
+        description: '[Proxy v1 API] :: Test for a valid request - External ID search',
+        successCriteria: 'Should return 200 with list of matching records with External ID',
+        scenario: 'Business',
```
Comment (utsabc) [inline test/integrations/destinations/salesforce/dataDelivery/other.ts:10]: This file should contain all the scenarios for network errors

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

File: test/integrations/destinations/salesforce/dataDelivery/other.ts (hunk 2 above)
Comment (yashasvibajpai) [reply test/integrations/destinations/salesforce/dataDelivery/other.ts]: moved to business

File: test/integrations/destinations/salesforce/dataDelivery/other.ts (hunk 3 above)
Comment (yashasvibajpai) [reply test/integrations/destinations/salesforce/dataDelivery/other.ts]: Addressed. Removed the file

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3074/test-report.html

File: test/integrations/destinations/salesforce/dataDelivery/other.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                  attemptNum: 1,
+                  userId: 'default-userId',
+                  destinationId: 'default-destinationId',
+                  workspaceId: 'default-workspaceId',
+                  sourceId: 'default-sourceId',
+                  secret: {
+                    accessToken: 'default-accessToken',
+                  },
+                  dontBatch: false,
+                },
```
Comment (utsabc) [inline test/integrations/destinations/salesforce/dataDelivery/other.ts:44]: A function can be used to reduce duplications

File: test/integrations/destinations/salesforce/dataDelivery/other.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            statTags: {
+              errorCategory: 'network',
+              errorType: 'retryable',
+              destType: 'SALESFORCE',
+              module: 'destination',
+              implementation: 'native',
+              feature: 'dataDelivery',
+              destinationId: 'default-destinationId',
+              workspaceId: 'default-workspaceId',
+            },
```
Comment (utsabc) [inline test/integrations/destinations/salesforce/dataDelivery/other.ts:56]: Declare constant and reduce duplications

**Review by utsabc — CHANGES_REQUESTED** [review]

**Review by utsabc — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**